data/model/gc.py

```diff
@@ -95,6 +118,28 @@ def purge_repository(repo, force=False):
     return True
 
 
+def _chunk_delete_all(repo, model, force=False, chunk_size=500):
+    """ Deletes all rows referencing the given repository in the given model. """
+    assert repo.state == RepositoryState.MARKED_FOR_DELETION or force
+
+    while True:
+        min_id = model.select(fn.Min(model.id)).where(model.repository == repo).scalar()
+        if min_id is None:
+            return
+
+        max_id = (
+            model.select(fn.Max(model.id))
+            .where(model.repository == repo, model.id <= min_id + chunk_size)
+            .scalar()
+        )
+        if min_id is None or max_id is None or min_id > max_id:
```

Comment:
```
if [...] min_id > max_id
```
Should we assert this is not true, as a sanity check? It shouldn't ever happen regardless because of your query but I do suppose things can change in the database after the `min_id` query was executed.

Reply:
Since this is just an optimization, if it fails, I'd rather just leave early than assert, since that would break the worker.